x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/total_feature_importane_summary/feature_importance_summary.tsx

```diff
@@ -0,0 +1,208 @@
+import { useMlKibana } from '../../../../../contexts/kibana';
+const { euiColorMediumShade } = euiVars;
+const axisColor = euiColorMediumShade;
+
+const axes: RecursivePartial<AxisStyle> = {
+  axisLine: {
+    stroke: axisColor,
+  },
+  tickLabel: {
+    fontSize: 12,
+    fill: axisColor,
+  },
+  tickLine: {
+    stroke: axisColor,
+  },
+  gridLine: {
+    horizontal: {
+      dash: [1, 2],
+    },
+    vertical: {
+      strokeWidth: 0,
+    },
+  },
+};
+const theme: PartialTheme = {
+  axes,
+};
+
+export interface FeatureImportanceSummaryPanelProps {
+  totalFeatureImportance: TotalFeatureImportance[];
+}
+
+const tooltipContent = i18n.translate(
+  'xpack.ml.dataframe.analytics.exploration.featureImportanceSummaryTooltipContent',
+  {
+    defaultMessage:
+      'Shows to what degree a given feature of a data point contributes to the prediction. The magnitude of feature importance shows how significantly the feature affects the prediction both locally (for a given data point) or generally (for the whole data set).',
+  }
+);
+
+export const FeatureImportanceSummaryPanel: FC<FeatureImportanceSummaryPanelProps> = ({
+  totalFeatureImportance,
+}) => {
+  const {
+    services: { docLinks },
+  } = useMlKibana();
+
+  const [plotData, barSeriesSpec] = useMemo(() => {
+    let sortedData: Array<{
+      featureName: string;
+      meanImportance: number;
+      className?: string;
+    }> = [];
+    let barSettings: Partial<BarSeriesSpec> = {
+      xAccessor: 'featureName',
+      yAccessors: ['meanImportance'],
+    };
+    if (totalFeatureImportance.length < 1) {
+      return [sortedData, barSettings];
+    }
+
+    if (isClassificationTotalFeatureImportance(totalFeatureImportance[0])) {
+      // if binary classification
+      if (totalFeatureImportance[0].classes.length === 2) {
+        sortedData = (totalFeatureImportance as ClassificationTotalFeatureImportance[]).map((d) => {
+          return {
+            featureName: d.feature_name,
+            // in case of binary classification, both classes will have the same mean importance
+            meanImportance: d.classes[0].importance.mean_magnitude,
+          };
+        });
+      }
+
+      // if multiclass classification
+      // stack them in order of increasing importance
+      // so for each feature, biggest importance on the left to smallest importance on the right
+      if (totalFeatureImportance[0].classes.length > 2) {
+        (totalFeatureImportance as ClassificationTotalFeatureImportance[]).forEach((feature) => {
+          feature.classes
+            .sort((a, b) => a.importance.mean_magnitude - b.importance.mean_magnitude)
+            .forEach((featureClass) => {
+              sortedData.push({
+                featureName: feature.feature_name,
+                meanImportance: featureClass.importance.mean_magnitude,
+                className: featureClass.class_name,
+              });
+            });
+        });
+        barSettings = {
```

Comment:
Is it possible to limit the height of a bar? When there are only two features for example, the bars look too big I think.

![image](https://user-images.githubusercontent.com/7405507/94163222-7e362200-fe7f-11ea-82ea-22ba6129e363.png)

Reply:
Updated here https://github.com/elastic/kibana/pull/78238/commits/c419c4223cb0609006dcae4b37197becc49b2176